addon/components/infinity-loader.js

```diff
@@ -205,11 +208,11 @@ const InfinityLoaderComponent = Component.extend(InViewportMixin, {
       let infinityModelContent = get(this, 'infinityModelContent');
 
       if (typeof(get(this, 'infinityLoad')) === 'function') {
-        // closure action
+        // closure action (if you need to perform some other logic)
         return get(this, 'infinityLoad')(infinityModelContent);
       } else {
-        // old action
-        this.sendAction('loadMoreAction', infinityModelContent);
+        // service action
+        get(this, 'infinity').infinityLoad(infinityModelContent, 1, this.element);
```

Comment:
Passing the element in the infinityLoad function will help when you have multiple infinity lists on your page.

Reply:
@hhff Is this where you think we can do the intelligent caching?

Reply:
Not quite - I was thinking we'd do labels:

```
model() {
  return this.infinityModel('product', {...options, label: 'MasterProductsCollection' })
}
```

```
{{infinity-loader label='MasterProductsCollection'}}
```


To do it without labels, we could also hashify (https://www.npmjs.com/package/object-hash) the argument signature to infinity model, like so:

```
model() {
  return this.infinityModel('product', {...options }) // infinity-service hashes these args and stores the result as `InfinityService.cache[hash]`, and returns that in the future
}
```

```
{{infinity-loader model=model}}
```

Reply:
Hashifying means that we can "recognize" infinity collections that look the same across the app, and not have to reload them from the API if that same collection is asked for in different routes. To do that with labels, the developer would need to use the same label in different routes (which is also fine)